src/binary-reader.cc

```diff
@@ -328,7 +328,7 @@ Result BinaryReader::ReadRefType(Type* out_value, const char* desc) {
   uint32_t type = 0;
   CHECK_RESULT(ReadS32Leb128(&type, desc));
   *out_value = static_cast<Type>(type);
-  ERROR_UNLESS(out_value->IsRef(), "%s must be a reference type", desc);
+  ERROR_UNLESS(out_value->IsFuncOrExternRef(), "%s must be a reference type", desc);
```

Comment:
Why not just leave this as `IsRef`?   Function refs are also refs, right?

Reply:
Thanks, in this case we indeed can use just IsRef